syntax = "proto3";

package dbibles;

import "google/protobuf/empty.proto";

// Message for a Bible Version
message BibleVersion {
    string name = 1;
}

// Message for a Bible Book
message BibleBook {
    string name = 1;
    int32 number = 2;
}

// Message for a Bible Chapter
message BibleChapter {
    int32 number = 1;
}

// Message for a Bible Verse
message BibleVerse {
    int32 number = 1;
    string content = 2;
}

// Service for accessing Bible data
service BibleService {
    // Get a list of available Bible versions
    rpc GetBibleVersions (google.protobuf.Empty) returns (BibleVersionList);

    // Get a list of books in a specific Bible version
    rpc GetBooksInVersion (BibleVersionRequest) returns (BibleBookList);

    // Get a list of chapters in a specific book
    rpc GetChaptersInBook (BibleBookRequest) returns (BibleChapterList);

    // Get the content of a specific verse
    rpc GetVerse (BibleVerseRequest) returns (BibleVerseResponse);
}

// Request message for getting Bible versions
message BibleVersionRequest {
    string version_name = 1;
}

// Response message for a list of Bible versions
message BibleVersionList {
    repeated BibleVersion versions = 1;
}

// Request message for getting books in a specific Bible version
message BibleBookRequest {
    string version_name = 1;
}

// Response message for a list of Bible books
message BibleBookList {
    repeated BibleBook books = 1;
}

// Request message for getting chapters in a specific book
message BibleChapterRequest {
    string version_name = 1;
    string book_name = 2;
}

// Response message for a list of Bible chapters
message BibleChapterList {
    repeated BibleChapter chapters = 1;
}

// Request message for getting the content of a specific verse
message BibleVerseRequest {
    string version_name = 1;
    string book_name = 2;
    int32 chapter_number = 3;
    int32 verse_number = 4;
}

// Response message for a Bible verse
message BibleVerseResponse {
    string content = 1;
}



























// syntax = "proto3";

// package bible;

// service BibleService {
//   rpc GetBibleVersions (EmptyRequest) returns (BibleVersionList);
//   rpc GetBooksInVersion (BibleVersionRequest) returns (BibleBookList);
//   rpc GetChaptersInBook (BibleChapterRequest) returns (BibleChapterList);
//   rpc GetVerseInChapter (BibleVerseRequest) returns (BibleVerseResponse);
//   rpc CompareVerses (CompareVersesRequest) returns (CompareVersesResponse);
// }

// message EmptyRequest {}

// message BibleVersionRequest {
//   string version = 1;
// }

// message BibleChapterRequest {
//   string version = 1;
//   string book = 2;
// }

// message BibleVerseRequest {
//   string version = 1;
//   string book = 2;
//   uint32 chapter = 3;
//   uint32 verse = 4;
// }

// message CompareVersesRequest {
//   repeated string versions = 1;
//   string book = 2;
//   uint32 chapter = 3;
//   uint32 verse = 4;
// }

// message BibleVersionList {
//   repeated BibleVersion versions = 1;
// }

// message BibleVersion {
//   string name = 1;
// }

// message BibleBookList {
//   repeated BibleBook books = 1;
// }

// message BibleBook {
//   string name = 1;
// }

// message BibleChapterList {
//   repeated uint32 chapters = 1;
// }

// message BibleVerseResponse {
//   string version = 1;
//   string book = 2;
//   uint32 chapter = 3;
//   uint32 verse = 4;
//   string content = 5;
// }

// message CompareVersesResponse {
//   repeated BibleVerseResponse verses = 1;
// }
